src/app/pages/Kafka/kafka-instance.ts

```diff
@@ -53,11 +73,60 @@ export const useKafkaInstance = (): KafkaInstance | false | undefined => {
 
   return kafkaRequest
     ? {
-        kafkaDetail: kafkaRequest as Required<KafkaRequest>,
-        adminServerUrl: getAdminServerUrl(
-          config.kafka?.adminServerUrlTemplate || DEFAULT_ADMIN_SERVER_URL_TEMPLATE,
-          kafkaRequest
-        ),
-      }
+      kafkaDetail: kafkaRequest as Required<KafkaRequestWithTopicConfig>,
+      adminServerUrl: getAdminServerUrl(
+        config.kafka?.adminServerUrlTemplate || DEFAULT_ADMIN_SERVER_URL_TEMPLATE,
+        kafkaRequest
+      ),
+    }
     : kafkaRequest;
 };
+
+/**
+ * Return list of the instance types available for the current user
+ *
+ * @returns {Promise<InstanceType[]>}
+ */
+
+export const useGetAvailableSizes = () => {
```

Comment:
Do we have any way to reuse code/avoid duplicating it across kas-ui and app-services-ui?